File: for-local-test/344.reverse-string.java
class Solution {
    public void reverseString(char[] s) {
        if (s == null || s.length <= 1) return;
        int left = 0;
        int right = s.length - 1;
        while (left < right){
            char temp = s[left];
            s[left] = s[right];
            s[right] = temp;
            left++;
            right--;
        }
    }
    public static void main(String[] args) {
        Solution solution = new Solution();
        char[] str1 = {'h', 'e', 'l', 'l', 'o'};
        solution.reverseString(str1);
        System.out.println(str1); // Output: "olleh"

        char[] str2 = {'H', 'a', 'n', 'n', 'a', 'h'};
        solution.reverseString(str2);
        System.out.println(str2); // Output: "hannaH"
    }
}

File: for-local-test/python/344.reverse-string.py
class Solution:
    def reverseString(self, s: list[str]) -> None:
        if s is None or len(s) <= 1:
            return
        left = 0
        right = len(s) - 1
        while left < right:
            s[left], s[right] = s[right], s[left]
            left += 1
            right -= 1

if __name__ == "__main__":
    solution = Solution()
    test_string = ['h', 'e', 'l', 'l', 'o']
    solution.reverseString(test_string)
    print(test_string)  # Output: ['o', 'l', 'l', 'e', 'h']